src/new_index/schema.rs

```diff
@@ -860,6 +882,76 @@ impl ChainQuery {
             |t| t == txid,
         ))
     }
+
+    /// Get MimbleWimble transaction outputs for given master private scan key
+    /// and spend pub keys.
+    pub fn get_mw_outputs(
+        &self, priv_scan_key: &[u8; 32], 
+        spend_pub_keys: Vec<bitcoin::secp256k1::PublicKey>
+    ) -> Vec<mimblewimble::Output> {
+        self.store.txstore_db.iter_scan(b"MWO")
+            .map(| row | { MWOutputRow::from_row(row).output })
+            .filter(| output | { identify_output(output, priv_scan_key, &spend_pub_keys) })
+            .collect_vec()
+    }
+}
+
+fn identify_output(
```

Comment:
This looks correct, but you'll probably want tests to verify. Once again, I'm willing to generate some test vectors for it, if you're willing to tolerate my general slowness.

Reply:
OK